plugins/scaffolder-backend/src/scaffolder/stages/prepare/azure.ts

```diff
@@ -22,24 +22,47 @@ import { InputError, Git } from '@backstage/backend-common';
 import { PreparerBase, PreparerOptions } from './types';
 import parseGitUrl from 'git-url-parse';
 import { Config } from '@backstage/config';
+import { Logger } from 'winston';
+import {
+  readAzureIntegrationConfigs,
+  AzureIntegrationConfig,
+} from '@backstage/integration';
 
 export class AzurePreparer implements PreparerBase {
-  private readonly privateToken: string;
+  private readonly integrations: AzureIntegrationConfig[];
+  private readonly scaffolderToken: string | undefined;
 
-  constructor(config: Config) {
-    this.privateToken =
-      config.getOptionalString('scaffolder.azure.api.token') ?? '';
+  constructor(config: Config, { logger }: { logger: Logger }) {
+    this.integrations = readAzureIntegrationConfigs(
+      config.getOptionalConfigArray('integrations.azure') ?? [],
+    );
```

Comment:
i will want to expose this type instead soon https://github.com/backstage/backstage/blob/master/packages/integration/src/types.ts#L42

so instead of dealing with the configs directly you would be able to do like `const integrations = readAzureIntegration(config)` and then `integrations.byHost(hostname).config.token` or whatever, so you don't have to do the lookups manually and knowing the config format for that part

Reply:
The idea (not this PR as it's huge) is that we will iterate through each of the integrations and create a new Publisher or Preparer for each host with the integration config so the preparers interface will go back to having a more simple interface injecting the client + tokens needed but we can't do that until everyone is off the scaffolder config hence this interim PR. 👍

Reply:
Ya what you are doing is fine, I'm just mentioning that in the near future i may make this even easier to consume. :)